File: app/src/main/java/com/example/clearentwrapperdemo/feature/HomeFragment.java
package com.example.clearentwrapperdemo.feature;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.example.clearentwrapperdemo.databinding.FragmentHomeBinding;

import java.util.function.Consumer;

public class HomeFragment extends Fragment {

    private FragmentHomeBinding binding;
    private final Runnable pairReaderHandler;
    private final Consumer<Double> startTransactionHandler;

    public HomeFragment(Runnable pairReaderHandler, Consumer<Double> startTransactionHandler) {
        this.pairReaderHandler = pairReaderHandler;
        this.startTransactionHandler = startTransactionHandler;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = FragmentHomeBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        setupViews();
    }

    private void setupViews() {
        binding.pairReaderButton.setOnClickListener(view -> pairReaderHandler.run());
        binding.startTransactionButton.setOnClickListener(view ->
                startTransactionHandler.accept(Double.parseDouble(binding.chargeAmountEditText.getText().toString()))
        );
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        binding = null;
    }
}

File: app/src/main/java/com/example/clearentwrapperdemo/feature/HomeFragment.kt
package com.example.clearentwrapperdemo.feature

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.example.clearentwrapperdemo.databinding.FragmentHomeBinding

class HomeFragment(
    private val pairReaderHandler: () -> Unit,
    private val startTransactionHandler: (Double) -> Unit
) : Fragment() {

    private var _binding: FragmentHomeBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentHomeBinding.inflate(inflater, container, false)

        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        setupViews()
    }

    private fun setupViews() {
        binding.apply {
            pairReaderButton.setOnClickListener {
                pairReaderHandler()
            }
            startTransactionButton.setOnClickListener {
                val chargeAmount: Double? = chargeAmountEditText.text?.toString()?.toDouble()

                chargeAmount?.also {
                    startTransactionHandler(it)
                }
            }
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}